### ISSUE TITLE
[Bug/Improvement] Incompatibility with Entware

### ISSUE THREAD

**Issue by OrpheeGT** [007revad/Synology_app_mover#125]
Hello,

I'm using entware on a daily basis, and it seems to be imcompatible with your script

it does not handle "sleep 0.3" and seems to manage filesystem size by %...

https://github.com/Entware/Entware/wiki/Install-on-Synology-NAS

I had to disable/unload entware from /etc/profile to run the script correctly.

**Comment by 007revad** [comment]
What do these commands return?
```
type bash
bash --version
```

**Comment by 007revad** [comment]
Are you using v.3.160 or v4.0.71 of the script?

Did the script stop after those errors? Or did it finish and they were the only errors?

Did you use Entware to install ash or dash?

I can't test my scripts with a #!/bin/ash shebang using shellcheck because when you set the shebang to #!/bin/ash it checks the script as if it was dash. I did check it with #!/bin/ash and #!/bin/sh 

- As #!/bin/ash there were 577 lines with errors (out of 3068 lines).
- As #!/bin/sh there were 609 lines with errors (out of 3068 lines).

The only time I use Entware is to install bash on my Asustor NAS because it's OS uses busybox and it's /bin/bash is a symlink to to ash and ash is a symlink to busybox... and busybox caused too many errors when I was making an Asustor version of my Synology Plex Backup script, like:
```
${var,,}		 variable expansion causes an error
|&			 piping stdout and stderr causes an error
2> >  and  >&2		 redirection causes an error
[[ "$var" != "a b" ]]	 causes an error
```

I used to use the #!/bin/bash shebang but it caused issues for edge cases like yours, so I changed it to `#!/sur/bin/env bash`. In your case changing the script's shebang to #!/bin/bash should solve your issue.

I'll change `sleep` to `/usr/bin/sleep` `du` to `/usr/bin/du`.

Or I could add: `export PATH=$(echo $PATH | sed -e 's/\/opt\/bin:\/opt\/sbin://')` near the top of the script.

**Comment by OrpheeGT** [comment]
I was using v4.0.71 

to be honest with the flood of errors about sleep, and sort in error too, I was not very confident about move consistancy.

But in the end it seemed it moved values anyway.

It seems busybox is a part of Entware default installation...

I tried to uninstall it, but it fails : 

```
root@NAS:~# opkg remove busybox
Refusing to remove essential package busybox.
        Removing an essential package may lead to an unusable system, but if
        you enjoy that kind of pain, you can force opkg to proceed against
        its will with the option: --force-removal-of-essential-packages
No packages removed.
```

I may try to force remove with -f ...

if you chose /usr/bin/ there is also "sort" to address.
Maybe the export part would be the best.

With busybox, everytime the script start, it checks for new version...

Apart from that, I don't think it is related, but I had an issue with FS type check for container manager...

Source and target were both ext4, and script always cancelled saying the source and target were not the same something like "source ext4 is not the same as target ext4"
I removed this check from the script to be able to make the move.
